dash/examples/ex.11.simple-stencil/main.cpp

```diff
@@ -0,0 +1,237 @@
+/**
+ * \example ex.11.simple-stencil/main.cpp
+ *
+ * Stencil codes are iterative kernels on arrays of at least 2 dimensions
+ * where the value of an array element at iteration i+1 depends on the values
+ * of its neighbors in iteration i.
+ *
+ * Calculations of this kind are very common in scientific applications, e.g.
+ * in iterative solvers and filters in image processing.
+ *
+ * This example implements a very simple blur filter. For simplicity
+ * no real image is used, but an image containg circles is generated.
+ * 
+ * \TODO fix \cdash::copy problem
+ */
+
+#include <dash/Init.h>
+#include <dash/Matrix.h>
+#include <dash/Dimensional.h>
+#include <dash/TeamSpec.h>
+#include <dash/algorithm/Fill.h>
+#include <dash/algorithm/Copy.h>
+
+#include <fstream>
+#include <string>
+#include <iostream>
+#include <vector>
+
+using namespace std;
+
+using element_t = unsigned char;
+using Array_t   = dash::NArray<element_t, 2>;
+using index_t = typename Array_t::index_type;
+
+
+void write_pgm(const std::string & filename, const Array_t & data){
+  if(dash::myid() == 0){
+
+    auto ext_x = data.extent(0);
+    auto ext_y = data.extent(1);
+    std::ofstream file;
+    file.open(filename);
+
+    file << "P2\n" << ext_x << " " << ext_y << std::endl;
+    // Data
+//    std::vector<element_t> buffer(ext_x);
+
+    for(long y=0; y<ext_y; ++y){
+      auto first = data.begin()+ext_x*y; 
+      auto last  = data.begin()+(ext_x*(y+1));
+
+//      BUG!!!!
+//      dash::copy(first, last, buffer.data());
+
+      for(long x=0; x<ext_x; ++x){
+//        file << buffer[x] << " ";
+        file << static_cast<int>(data[x][y]) << " ";
+      }
+      file << std::endl;
+    }
+    file.close();
+  }
+  dash::barrier();
+}
+
+void set_pixel(Array_t & data, index_t x, index_t y){
+  const element_t color = 1;
+  auto ext_x = data.extent(0);
+  auto ext_y = data.extent(1);
+
+  x = (x+ext_x)%ext_x;
```

Comment:
I stumbled upon this too. X might be negative and hence I have to use modulo twice to simulate the torus surface.